src/cmake/golang.cmake

```diff
@@ -0,0 +1,32 @@
+function(add_go_library TARGET SOURCE)
```

Comment:
header?

Reply:
The header file is added as a generated source artifact at lines 17-18, the caller only needs to provide the source `*.go` file to this function.